serialization/class_definition.go

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package serialization
+
+import "github.com/hazelcast/hazelcast-go-client/hzerrors"
+
+// ClassDefinition defines a class schema for Portable structs.
+type ClassDefinition struct {
```

Comment:
One thing we lost with this change is that, with the builder approach, we know that the class definition is immutable and correct. We might lose parts of these guarantees in exchange for a simpler API. It might be OK to do so, but I just wanted to write this

Reply:
Thanks, `ClassDefinition` is copied before actually used internally, so user modifications don't matter.